setup.sh

```diff
@@ -0,0 +1,617 @@
+#!/bin/bash
+
+# list of colors
+# check if stdout is a terminal...
+if test -t 1; then
+  # see if it supports colors...
+  ncolors=$(tput colors)
+  if test -n "$ncolors" && test $ncolors -ge 8; then
+    BOLD=$(tput bold)
+    RED="$(tput setaf 1)"
+    BLUE="$(tput setaf 4)"
+    GREEN="$(tput setaf 2)"
+    YELLOW="$(tput setaf 3)"
+    NORMAL="$(tput sgr0)"
+  fi
+fi
+
+CURRENT_OS=Unknown
+OS_IS_SUPPORTED=0
+DOCKER_COMPOSE_BIN=/usr/local/bin/docker-compose
+ANALYTICS_URL="http://analytics.kuzzle.io/"
+GITTER_URL="https://gitter.im/kuzzleio/kuzzle"
+SUPPORT_MAIL="support@kuzzle.io"
+
+# Output a text with the selected color (reinit to normal at the end)
+write() {
+  echo -e " $1$2" "$NORMAL"
+}
+
+writeBold() {
+  echo -e "${BOLD} $1$2" "$NORMAL"
+}
+
+promptBold() {
+  echo -n -e "${BOLD} $1$2" "$NORMAL"
+}
+
+commandExists() {
+  command -v "$@" > /dev/null 2>&1
+}
+
+vercomp () {
+  if [[ $1 == $2 ]]
+  then
+    return 0
+  fi
+  local IFS=.
+  local i ver1=($1) ver2=($2)
+  # fill empty fields in ver1 with zeros
+  for ((i=${#ver1[@]}; i<${#ver2[@]}; i++))
+  do
+    ver1[i]=0
+  done
+  for ((i=0; i<${#ver1[@]}; i++))
+  do
+    if [[ -z ${ver2[i]} ]]
+    then
+      # fill empty fields in ver2 with zeros
+      ver2[i]=0
+    fi
+    if ((10#${ver1[i]} > 10#${ver2[i]}))
+    then
+      return 1
+    fi
+    if ((10#${ver1[i]} < 10#${ver2[i]}))
+    then
+      return 2
+    fi
+  done
+  return 0
+}
+
+findOSType()
+{
+  osType=$(uname)
+  if [ "$osType" == "Linux" ]; then
+    if [ -f /etc/os-release ]; then
+      distro=$(awk -F= '/^NAME/{print $2}' /etc/os-release)
+    else
+      distro=$(ls -d /etc/[A-Za-z]*[_-][rv]e[lr]* | grep -v "lsb" | cut -d'/' -f3 | cut -d'-' -f1 | cut -d'_' -f1)
+    fi
+    CURRENT_OS=$(echo $distro | tr 'a-z' 'A-Z' | cut -d' ' -f1 | tr -d '"')
+  fi
+}
+
+isOSSupported()
+{
+  case "$CURRENT_OS" in
+    "UBUNTU" | "DEBIAN")
+    {
+      OS_IS_SUPPORTED=1
+    } ;;
+    *)
+    {
+      echo
+      writeBold "$YELLOW" "[✖] Your OS ($CURRENT_OS) is not officially supported."
+      write               "    This means we didn't thoroughly test Kuzzle on your"
+      write               "    system. It is likely to work, so you may want to continue."
+      OS_IS_SUPPORTED=0
+      while [[ "$proceedNotSupported" != [yYnN] ]]
```

Comment:
still think it is a shame to block the installation process where there are chances it would work but ok...